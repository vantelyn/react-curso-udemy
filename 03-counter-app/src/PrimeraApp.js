//  FUNCTIONAL COMPONENTS (APPS BASADAS EN FUNCIONES QUE HACEN USO DE HOOKS PARA SIMPLIFICAR EL CÓDIGO)
import React from 'react';
// import { Fragment } from 'react';   
import PropTypes from 'prop-types';


const PrimeraApp = ({ saludo, subtitulo }) =>{

    // if(!saludo){
    //     throw new Error('El saludo es necesario');
    // }



    return (
        // <Fragment>            
        //     <h1> { saludo } </h1>
        //     <p> { subtitulo } </p>
        // </Fragment>
        <>            
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
            <h1>{ saludo }!!!</h1>
            <p>{ subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes ={
    saludo: PropTypes.string.isRequired,
    // otra: PropTypes.number,
}

PrimeraApp.defaultProps = {
    subtitulo: 'Subtítulo por defeco',
}

export default PrimeraApp;