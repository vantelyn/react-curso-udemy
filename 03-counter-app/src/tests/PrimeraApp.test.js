// import { render } from "@testing-library/react";
import React from "react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('Pruebas en <PrimeraApp />', () => { 
    
    // test('Debe de mostrar el mensaje "Hola, soy Germán"', () => { 
    //     const saludo = 'Hola. soy Germán';
    //     const { getByText } = render( <PrimeraApp saludo= { saludo } /> );
    //     expect( getByText( saludo ) ).toBeInTheDocument(); 
    //  })


    test('debe de mostrar <PrimeraApp /> correctamente', () => { 

        const saludo = 'Hola, soy Germán';        

        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar el subtítulo enviado por props', () => { 

        const saludo = 'Hola, soy Germán';
        const subtitulo = 'Me gustan las gominolas';

        const wrapper = shallow( <PrimeraApp saludo={ saludo } subtitulo={ subtitulo } /> );

        const textoParrafo = wrapper.find('p').text();
        
        expect(textoParrafo).toBe(subtitulo);

     });

});