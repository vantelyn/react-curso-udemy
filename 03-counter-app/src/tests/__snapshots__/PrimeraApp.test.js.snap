// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Germán
    !!!
  </h1>
  <p>
    Subtítulo por defeco
  </p>
</Fragment>
`;
